File: CS/WindowsApplication1/CustomControls/CustomCheckedListBoxControl/CustomCheckedListBoxViewInfo.cs
using System;
using System.Drawing;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;

namespace Example
{
    public class CustomCheckedListBoxViewInfo : CheckedListBoxViewInfo
    {
        public CustomCheckedListBoxViewInfo(CheckedListBoxControl listBox) : base(listBox) { }
        private CustomCheckedListBoxControl ListBox { get { return OwnerControl as CustomCheckedListBoxControl; } }

        protected override ItemInfo CreateItemInfo(Rectangle bounds, object item, string text, int index)
        {
            ItemInfoEventArgs e = new ItemInfoEventArgs(index, item);
            ListBox.RaiseShiftItem(e);
            Rectangle itemBounds = new Rectangle(bounds.X + e.ItemOffset, bounds.Y, bounds.Width, bounds.Height);
            ItemInfo itemInfo = base.CreateItemInfo(itemBounds, item, text, index);
            itemInfo.Bounds = bounds;
            return itemInfo;
        }


    }
}

File: VB/WindowsApplication1/CustomControls/CustomCheckedListBoxControl/CustomCheckedListBoxViewInfo.vb
Imports System
Imports System.Drawing
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo

Namespace Example
	Public Class CustomCheckedListBoxViewInfo
		Inherits CheckedListBoxViewInfo

		Public Sub New(ByVal listBox As CheckedListBoxControl)
			MyBase.New(listBox)
		End Sub
		Private ReadOnly Property ListBox() As CustomCheckedListBoxControl
			Get
				Return TryCast(OwnerControl, CustomCheckedListBoxControl)
			End Get
		End Property

		Protected Overrides Function CreateItemInfo(ByVal bounds As Rectangle, ByVal item As Object, ByVal text As String, ByVal index As Integer) As ItemInfo
			Dim e As New ItemInfoEventArgs(index, item)
			ListBox.RaiseShiftItem(e)
			Dim itemBounds As New Rectangle(bounds.X + e.ItemOffset, bounds.Y, bounds.Width, bounds.Height)
			Dim itemInfo As ItemInfo = MyBase.CreateItemInfo(itemBounds, item, text, index)
			itemInfo.Bounds = bounds
			Return itemInfo
		End Function


	End Class
End Namespace